src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -321,6 +321,99 @@ def validate_select(namespace):
 
 # region COMMAND VALIDATORS
 
+
+def process_download_batch_parameters(namespace):
+    """Process the parameters for storage blob download command"""
+    from azure.cli.command_modules.storage.storage_url_helpers import parse_url
+
+    # 1. quick check
+    if namespace.destination is None:
+        raise ValueError('Destination parameter is missing.')
+
+    if namespace.source is None:
+        raise ValueError('Source parameter is missing.')
+
+    if not os.path.exists(namespace.destination) or not os.path.isdir(namespace.destination):
+        raise ValueError('Destination folder {} does not exist'.format(namespace.source))
```

Comment:
this is the choice i made to reduce the disk operation this command will execute so as to simply the logic.